src/router.js

```diff
@@ -61,6 +61,13 @@ const checkAndSubscribeToLobby = async (to) => {
   }
 };
 
+const getGameState = (to) => {
+  const gameStore = useGameStore();
+  const gameId = parseInt(to.params.gameId);
+  const gameStateIndex = parseInt(to.query.gameStateIndex ?? -1);
+  return gameStore.requestGameState(gameId, gameStateIndex);
+};
```

Comment:
I like this idea, can we do it in spectate as well?

Reply:
The main difference is that spectate also should create a `UserSpectatingGame` entry for the requesting user. We currently do that + get the latest game data via the `/spectate` endpoint. It may make sense to combine those to parameterize one endpoint to handle both cases, but for now keeping them separate is simpler